### PR TITLE
Fix some issues of the digitalSTROM-Binding

### LINKED ISSUE #5554: digitalstrom binding... more issues
I use openHAB 2.2 with digitalstrom binding 2.2.0. The position of my blinds is not reliably reported between digitalstrom and openHAB. If I issue a command in OH then the position follows in OH. If I issue a command via dSS then OH does not update. And another "problem" in this context. dSS uses the value 100 as closed (weird, IMO) and 0 for open. The binding does the opposite. Now, strange enough, if I send a command 70 via binding I do get 30 in dSS which is what I intended to have. But it is anyway somewhat awkward, isn't it?

### LINKED ISSUE #5651: [digitalstrom] OpenHAB 2.3
Ever since I upgraded to openhab 2.3.0-1 the digitalstrom binding ceased to work.. 
None of the devices accepts any command anymore. Instead, the status in paperUI is reported "ONLINE - CONFIGURATION_PENDING waiting for listener registration" forever. A couple of reboots did not help. Things worked quite fine until yesterday. Does anyone notice the same behavior?

### PR DESCRIPTION
**Pull request by MichaelOchel**
* Change BaseTemperatureControl#isConfigured() to isNotSetOff(), due to changes of the dS-API (the parameter is removed  now). The method use the ControlMode#OFF to identify, if the temperature-control is activated (configured), now. I hope that the climate-control will continue to work. Fixed #5651
* add a note to the readme, that point out that directly set output values are not recognized. Fixed #5554

@maieralex maybe you can test the first point.

Also-by: Matthias Siegele <digitalSTROM-ESH-binding@web.de>
Signed-off-by: Michael Ochel <michael.ochel@koeln.de>

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/README.md [hunk 1]
```diff
@@ -374,6 +374,10 @@ The digitalSTROM-Server
 
 **Notes: **
 
+*Generally:*
+
+* The dSS only inform about scene-commands. So if you set the output value of devices e.g. through  the dSS-App, the binding will not be informed about the changes and you have to send a "refresh-command" to update the channel.
```
Comment (triller-telekom) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/README.md]: Who is "The dSS" here? And whom does it inform?

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/climate/jsonresponsecontainer/BaseTemperatureControl.java [hunk 2]
```diff
@@ -94,12 +83,13 @@ public Short getControlMode() {
     }
 
     /**
-     * Returns true, if heating for this zone is configured, otherwise false.
+     * Returns true, if heating for this zone is not set off (set {@link ControlModes} = {@link ControlModes#OFF}),
+     * otherwise false.
      *
-     * @return the isConfigured
+     * @return the true, if the set {@link ControlModes} is not {@link ControlModes#OFF}
```
Comment (triller-telekom) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/climate/jsonresponsecontainer/BaseTemperatureControl.java]: `@returns true, if the...`

**Comment by kaikreuzer** [comment]
Thanks for your fixes @MichaelOchel! If I get you right, you do not have the possibility for testing yourself - how shall we proceed with the PR then? Merge it untested? Or wait for any DS user (@maieralex, @maxx92, @Torkalis, @hmerk) to confirm the fixes to work?

**Comment by Torkalis** [comment]
Thanks for fixing but maybe i'm to stupid today... how can i test it to help you? Download?

**Comment by sjka** [comment]
Here's the binding: 

[org.eclipse.smarthome.binding.digitalstrom_0.10.0.201806201606.zip](https://github.com/eclipse/smarthome/files/2119649/org.eclipse.smarthome.binding.digitalstrom_0.10.0.201806201606.zip)

I exported it from the IDE using the latest commit in this PR (i.e. [92e9b73](https://github.com/eclipse/smarthome/pull/5673/commits/92e9b739f167ea4a021ef8b0416c53e09df5c9e6)). Only I had to rename it to `.zip` in order to make GitHub happy. Just rename it back to `.jar` and then update your bundle with it.

**Comment by maxx92** [comment]
I just installed it by copying it into directory openHAB-share>openhab-addon

Items come back to life as far as I can tell. However, one thing is strange. There are digitalstrom items that update now by the minute
'2018-06-20 16:55:53.256 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 16:56:57.306 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 16:57:58.222 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 16:59:04.427 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:00:10.225 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:01:11.080 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:02:11.957 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:03:14.321 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:04:18.403 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:05:19.304 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:06:20.155 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:07:24.244 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:08:27.870 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:09:33.550 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:10:37.231 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:11:41.316 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:12:42.651 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:13:46.654 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:14:49.320 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:15:57.343 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:17:00.296 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:18:05.352 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:19:09.429 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
'

**Comment by maxx92** [comment]
`2018-06-20 16:55:53.256 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 16:56:57.306 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 16:57:58.222 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 16:59:04.427 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:00:10.225 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:01:11.080 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:02:11.957 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:03:14.321 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:04:18.403 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:05:19.304 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:06:20.155 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:07:24.244 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:08:27.870 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:09:33.550 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:10:37.231 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:11:41.316 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:12:42.651 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:13:46.654 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:14:49.320 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:15:57.343 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:17:00.296 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:18:05.352 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.
2018-06-20 17:19:09.429 [me.event.ThingUpdatedEvent] - Thing 'digitalstrom:GE:302ed89f43f02ba000010414:303505d7f8000040000417a2' has been updated.`

**Comment by maxx92** [comment]
Hello all. A quick update on the latest binding for digitalstrom. The biggest problem is that I don't get any status back from any of the digitalstrom things! They just no longer communicate anything to OH. Therefore, no rule that uses status of the things will work anymore. However, I mostly use Homematic things to trigger rules and the digitalstrom things still take the commands. Just not the other way around. You could say that the binding has lost 50% of its potential. If you need more specific information from me to fix that, let me know.

**Comment by maxx92** [comment]
`2018-06-23 16:18:29.827 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception
java.lang.IllegalArgumentException: Value must be between 0 and 100
	at org.eclipse.smarthome.core.library.types.PercentType.validateValue(PercentType.java:55) [101:org.eclipse.smarthome.core:0.10.0.201805241348]
	at org.eclipse.smarthome.core.library.types.PercentType.<init>(PercentType.java:40) [101:org.eclipse.smarthome.core:0.10.0.201805241348]
	at org.eclipse.smarthome.binding.digitalstrom.handler.DeviceHandler.onDeviceStateChanged(DeviceHandler.java:396) [237:org.eclipse.smarthome.binding.digitalstrom:0.10.0.201806201606]
	at org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.impl.DeviceImpl.informListenerAboutStateUpdate(DeviceImpl.java:1669) [237:org.eclipse.smarthome.binding.digitalstrom:0.10.0.201806201606]
	at org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.impl.DeviceImpl.updateInternalDeviceState(DeviceImpl.java:1516) [237:org.eclipse.smarthome.binding.digitalstrom:0.10.0.201806201606]
	at org.eclipse.smarthome.binding.digitalstrom.internal.lib.sensorjobexecutor.sensorjob.impl.DeviceOutputValueSensorJob.execute(DeviceOutputValueSensorJob.java:75) [237:org.eclipse.smarthome.binding.digitalstrom:0.10.0.201806201606]
	at org.eclipse.smarthome.binding.digitalstrom.internal.lib.sensorjobexecutor.AbstractSensorJobExecutor$ExecutorRunnable.run(AbstractSensorJobExecutor.java:73) [237:org.eclipse.smarthome.binding.digitalstrom:0.10.0.201806201606]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:?]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]`

**Comment by kaikreuzer** [comment]
I'd suggest to merge this as it seems to fix at least a few things - I'll include it in a new openHAB distro by tonight, so people can easily test it. Anything that is then still open should be reported as a new issue.

**Comment by maxx92** [comment]
Do I have to delete "org.eclipse.smarthome.binding.digitalstrom_0.10.0.201806201606.zip"
 from my addon folder?

**Comment by kaikreuzer** [comment]
Yes, and you should never have had a zip file in there in the first place...

**Comment by maxx92** [comment]
.. but i renamed it "jar" as per instruction..

**Comment by maxx92** [comment]
what is the proper procedure now to have this done properly? I delete the file in openhab2-addons (it is the only one there) and then? run the openhabian-config? I am sorry but this is beyond my expertise level

**Comment by triller-telekom** [comment]
@maxx92 Simplest solution: Remove that file, update to the latest OH snapshot (this contains the updated binding) and restart OH.

**Comment by kaikreuzer** [comment]
If it is a .jar, then yes, please remove it from addons - while the instance is running as otherwise, the bundle isn't deinstalled and you'd have to do that manually (see [here](https://www.openhab.org/docs/administration/bundles.html#list-bundles)).

**Comment by maxx92** [comment]
Hello all. Actual digitalstrom server software 1.14.3. With this software the binding functionality is very reduced. I would like to update the binding but I have no programming background since dBase III.. Is there still hope for an adaption of the binding to the latest dss software? I am also still very interested in getting my shade brick GR-HKL230 supported. Thank you.

**Comment by maxx92** [comment]
addon: I am available for any testing or supplying logs..

**Comment by triller-telekom** [comment]
@maxx92 Unfortunately only @MichaelOchel is working on this binding if he finds the time to do so. But maybe you can motivate him by placing a bounty on your features like other people did for openHAB features here: https://www.bountysource.com/?search=openhab :)

**Comment by cschnider** [comment]
Hi all. I'm also running a 1.14.3 dS system and most current openHAB snapshot release. 
In the past binding was running smoothly without many issues, but know I'm experiencing also the above problems.

Is there any plan to adapt the binding in the near future? I know that openHAB and all of its components is very dependent on volunteers. I'm also available for testing and supplying logs.

Thank you very much for your help!

**Pull request merged**